pkg/pmem-csi-operator/controller/deployment/controller_driver.go

```diff
@@ -93,6 +95,9 @@ func (d *PmemCSIDriver) Reconcile(r *ReconcileDeployment) (bool, error) {
 	klog.Infof("Deployment: %q, state %q, changes %v, in cache %v", d.Name, d.Status.Phase, changes, foundInCache)
 
 	requeue, err := d.reconcileDeploymentChanges(r, changes, foundInCache)
+	// Revert any spec changes(like deployment defaults) we made.
+	// Those are not supposed to save on the API server.
+	d.Deployment.Spec = copy.Spec
```

Comment:
Is this still needed?

Reply:
Yes, it's because all the clients do not support patching status reliability. In the context of unit tests, the test client does not support patching status instead it updates the whole object, which leads to unexpected failures. So to cover all cases we revert spec changes.

Reply:
It would be useful to explain that in the code because it isn't obvious.

Reply:
Okay, added that to the comment.